hazelcast/src/main/java/com/hazelcast/internal/cluster/impl/ClusterServiceImpl.java

```diff
@@ -306,6 +306,7 @@ public void merge(Address newTargetAddress) {
     public void reset() {
         lock.lock();
         try {
+            convertLocalMemberToLite();
```

Comment:
this method is also called during force-start. I suspect that converting the member to a local member here is breaking the hot restart process.

Reply:
fixed inside cluster-merge-task